google-cloud-bigquery/lib/google/cloud/bigquery/policy.rb

```diff
@@ -0,0 +1,390 @@
+        #     group is recovered, this value reverts to `group:<emailid>` and the recovered group retains the role in
+        #     the binding.
+        #   * `domain:<domain>`: The G Suite domain (primary) that represents all the users of that domain. For example,
+        #     `google.com` or `example.com`.
+        #
+        # @return [Binding, nil] The binding object that was added to the policy. This object is mutable and may be used
+        #   to add or remove members within the context of a block passed to {Table.update_policy}.
+        #
+        # @example Update a mutable policy with {Table#update_policy}.
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   table = dataset.table "my_table"
+        #
+        #   table.update_policy do |p|
+        #     p.set_binding "roles/viewer", "user:viewer@example.com"
+        #   end # 2 API calls
+        #
+        def set_binding role, members
+          new_binding = Binding.new role, members
+          @bindings[role] = new_binding
+          new_binding
+        end
+
+        ##
+        # Convenience method deleting a binding in the policy. Returns `nil` if no binding is present for the role. See
+        # [Understanding Roles](https://cloud.google.com/iam/docs/understanding-roles) for a list of primitive and
+        # curated roles. See [BigQuery Table ACL
+        # permissions](https://cloud.google.com/bigquery/docs/table-access-controls-intro#permissions) for a list of
+        # values and patterns for members.
+        #
+        # @param [String] role A role that is bound to members in the policy. For example, `roles/viewer`,
+        #   `roles/editor`, or `roles/owner`. Required.
+        #
+        # @return [Binding, nil] The frozen binding object, or `nil` if no binding exists for the given role.
+        #
+        # @example Update a mutable policy with {Table#update_policy}.
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   table = dataset.table "my_table"
+        #
+        #   table.update_policy do |p|
+        #     p.remove_binding "roles/owner"
+        #   end # 2 API calls
+        #
+        def remove_binding role
+          @bindings.delete(role).freeze
+        end
+
+        ##
+        # @private Convert the Policy to a Google::Apis::BigqueryV2::Policy.
+        def to_gapi
+          Google::Apis::BigqueryV2::Policy.new(
+            bindings: bindings_to_gapi,
+            etag:     etag,
+            version:  1
+          )
+        end
+
+        ##
+        # @private Deep freeze the policy including its bindings.
+        def freeze
+          super
+          @bindings.values.map(&:freeze)
```

Comment:
Suggest `each` instead of `map` if we're not using the return value. (And similar in more cases below in this file.)

Reply:
Correct, thanks for catching this. Done.